### PR TITLE
V0.6.1 rc

### PR DESCRIPTION
**Pull request by kennyweiss**
# Summary

- This PR is a release candidate
- I created a helper PR (#720) to make it easier to see the diffs in this branch w.r..t `develop`, but please review this one.

### CODE REVIEW

**Review by white238 — APPROVED** [review]

**Comment by agcapps** [comment]
OK, I'll look over #720 but put comments and approvals on this PR.

**Review by agcapps — APPROVED** [review]
Yes, looks good.

**Review by rhornung67 — APPROVED** [review]

**Review by bmhan12 — APPROVED** [review]

**Review by nselliott — APPROVED** [review]

File: RELEASE-NOTES.md [hunk 1]
```diff
@@ -651,7 +653,8 @@ The Axom project release numbers follow [Semantic Versioning](http://semver.org/
 ###  Security
 - Use this section in case of vulnerabilities
 
-[Unreleased]:    https://github.com/LLNL/axom/compare/v0.6.0...develop
+[Unreleased]:    https://github.com/LLNL/axom/compare/v0.6.1...develop
+[Version 0.6.0]: https://github.com/LLNL/axom/compare/v0.6.0...v0.6.1
```
Comment (ltaylor16) [inline RELEASE-NOTES.md:657]: Should be `[Version 0.6.1]`.

**Review by ltaylor16 — APPROVED** [review]

File: RELEASE-NOTES.md (hunk 1 above)
Comment (kennyweiss) [reply RELEASE-NOTES.md]: Thanks @ltaylor16 -- fixed!

**Pull request merged**